tests/interfaces/records/handlers/records-write.spec.ts

```diff
@@ -1122,56 +1122,6 @@ describe('RecordsWriteHandler.handle()', () => {
         expect(reply.status.detail).to.contain(`no allow rule defined for Write`);
       });
 
-      it('should fail authorization if path to expected recipient in definition is longer than actual length of ancestor message chain', async () => {
```

Comment:
Removing this test because
1. Taking the example of `emailProtocol`, we allow `anyone: { "to": ["write"] }` and allow `author: { "of": "email", "to": ["read"] }`. When we write an email, we scan through (`anyone`, `author`, `recipient`). For the initial `write`, the `author` rule will have a longer expected path (`email`) than it has length of ancestor chain (length == 0). In this case, we shouldn't throw an error because this is a reasonable case. Instead, we should simply not add the `read` to allowed actions.
2. This test feels like its testing implementation more than expected behavior. I find it kinda confusing. If I were a developer using `dwn-sdk-js`, I don't think I would find it more helpful than an error saying `read is not an allowed action`.